twake/frontend/src/app/scenes/Client/ChannelsBar/NewDirectMessagesPopup.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { FC, useState } from 'react';
+
+import Languages from 'services/languages/languages.js';
+import Button from 'components/Buttons/Button.js';
+import ChannelsService from 'services/channels/channels.js';
+import MediumPopupComponent from 'app/services/Modal/ModalManager';
+import { ObjectModal, ObjectModalTitle } from 'components/ObjectModal/ObjectModal.js';
+import UserListManager from 'components/UserListManager/UserListManager.js';
+import { ChannelType, ChannelResource } from 'app/models/Channel';
+import Collections from 'app/services/CollectionsReact/Collections';
+
+import { useParams } from 'react-router-dom';
+import RouterServices from 'services/RouterServices';
+import OldCollections from 'services/Depreciated/Collections/Collections';
+
+const NewDirectMessagesPopup: FC = () => {
+  const [newUserDiscussion, setNewUserDiscussion] = useState<string[]>([]);
+
+  // TODO find a better way to get company id
+  const params: any = useParams();
+  const workspaceId = RouterServices.translateToUUID(params.workspaceId);
+  const company_id = OldCollections.get('workspaces').find(workspaceId)?.group?.id;
+
+  const collectionPath: string = `companies/${company_id}/workspaces/direct/channels/`;
+  const ChannelsCollections = Collections.get(collectionPath);
+
+  const upsertDirectMessage = async (): Promise<any> => {
+    const newDirectMessage: ChannelType = {
+      company_id: company_id,
+      workspace_id: workspaceId,
+      visibility: 'direct',
+    };
+
+    await ChannelsCollections.upsert(new ChannelResource(newDirectMessage), {
+      httpOptions: { members: newUserDiscussion },
+    });
+    await ChannelsService.openDiscussion(newUserDiscussion);
```

Comment:
Please comment this, it calls the old APIs, so it's cheating ;)

Reply:
Commented :joy: